relation-cache-updater/src/main/scala/org/sunbird/job/relationcache/functions/RelationCacheUpdater.scala

```diff
@@ -59,14 +59,16 @@ class RelationCacheUpdater(config: RelationCacheUpdaterConfig)
             val rootId = event.identifier
             logger.info("Processing - identifier: " + rootId)
             val hierarchy = getHierarchy(rootId)(metrics)
-            if (MapUtils.isNotEmpty(hierarchy)) {
-                val leafNodesMap = getLeafNodes(rootId, hierarchy)
+            if (StringUtils.isNotBlank(hierarchy)) {
+                storeHierarchyData(rootId, hierarchy, dataCache) (metrics)
+                val hierarchyMap = mapper.readValue(hierarchy, classOf[java.util.Map[String, AnyRef]])
+                val leafNodesMap = getLeafNodes(rootId, hierarchyMap)
```

Comment:
not sure why we need to move the code ? any reasons ? getHierarchy was already validating this right ?